app/controllers/middleware_deployment_controller.rb

```diff
@@ -17,4 +28,48 @@ def show
     show_container(@record, controller_name, display_name)
   end
 
+  def button
+    selected_operation = params[:pressed].to_sym
+    if OPERATIONS.key?(selected_operation)
```

Comment:
Yes, the code is very similar to the one from server controller, although parts of it have it's own specifics. It's worth identifying which parts make sense to generalize so they work for both.

Any idea of a good place where to create the helper ?